## Quantification of UNC13A splice variants
RNA-Seq data generated by NYGC ALS Consortium cohort were downloaded from the NCBI’s Gene Expression Omnibus (GEO) database (GSE137810, GSE124439, GSE116622, and GSE153960). The pre-processing and reads alignment (hg38) were done as described in RNA_alignment_and_splicing_analysis.  
1. PCR duplicates were removed using MarkDuplicates from Picard Tools (2.23.0) using the command `MarkDuplicates REMOVE_DUPLICATES=true CREATE_INDEX=true`.  
2. Only uniquely mapped reads were retained using the command `samtools view -b -q 255`.  
3. Reads that span either “Exon 19-Exon 20” junction, “Exon 20-CE” junction, “CE-Exon 21” junction, or “Exon 20-exon 21” junction were quantified using bedtools (2.27.1) as shown in junction_spanning_reads.sh. The required bed files are in the bed_files folder.  
